### PR TITLE
types: add annotations to keybind lib

### PR DESCRIPTION
**Pull request by aarnphm**
Adding annotations for keybind lib for ease-of-use

Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

### CODE REVIEW

File: lua/keymap/bind.lua [hunk 1]
```diff
@@ -18,98 +28,128 @@ function rhs_options:new()
...
+---@param mapping table<string, RhsContainer>
 function pbind.nvim_load_mapping(mapping)
 	for key, value in pairs(mapping) do
 		local mode, keymap = key:match("([^|]*)|?(.*)")
 		if type(value) == "table" then
 			local rhs = value.cmd
 			local options = value.options
 			local buf = value.buffer
-			if buf then
+			if buf and type(buf) == "number" then
```
Comment (aarnphm) [inline lua/keymap/bind.lua:152]: make sure that buf is number here, since buf can be boolean

`nvim_buf_set_keymap` takes integer only

**Review by Jint-lzxy — APPROVED** [review]
Made several minor modifications. LGTM 🍑

File: lua/keymap/bind.lua [hunk 2]
```diff
@@ -18,98 +28,128 @@ function rhs_options:new()
...
 
+---@param callback fun():nil
+---@return map_rhs
 function pbind.map_callback(callback)
 	local ro = rhs_options:new()
 	return ro:map_callback(callback)
 end
 
+---@param mapping table<string, map_rhs>
 function pbind.nvim_load_mapping(mapping)
```
Comment (CharlesChiuGit) [inline lua/keymap/bind.lua:145]: Please check the latest commits. It's not ``pbind` now.

**Review by CharlesChiuGit — CHANGES_REQUESTED** [review]

File: lua/keymap/bind.lua (hunk 2 above)
Comment (CharlesChiuGit) [reply lua/keymap/bind.lua:145]: omg @Jint-lzxy u move to fast lol

Comment (CharlesChiuGit) [reply lua/keymap/bind.lua:145]: Sorry! My fault! I mistake this with the drafting PR.
False alarm!

**Pull request merged**